jenkins_jobs/modules/project_matrix.py

```diff
@@ -162,9 +162,10 @@ def root_xml(self, data):
         strategy = data.get(strategy_name, {})
 
         if strategy_name == "execution-strategy":
-            XML.SubElement(root, "combinationFilter").text = str(
-                strategy.get("combination-filter", "")
-            ).rstrip()
+            if strategy.get("combination-filter", None):
```

Comment:
the code below checks for "touchstone" key like this:

    if `touchstone` in strategy:

then here it should be like:

    if 'combination-filter' in strategy: